### PR TITLE
s3Driver allows writing in folders

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

**Review by svenngjeto — APPROVED** [review]

File: s3/src/test/java/no/unit/nva/s3/S3DriverTest.java [hunk 1]
```diff
@@ -180,6 +184,23 @@ public void insertAndCompressFilesFilesStoresAllFilesCompressedInGzipStream() th
         assertThat(actualContent, is(equalTo(input)));
     }
 
+    @ParameterizedTest(name = "insertAndCompressFilesStoresAllFilesCompressedInGzipStreamInSpecifiedFolder")
+    @ValueSource(strings = {EMPTY_STRING, ROOT})
```
Comment (torarnet) [inline s3/src/test/java/no/unit/nva/s3/S3DriverTest.java:188]: Im not so experienced in using ParameterizedTest but how are you using 2 strings, `EMPTY_STRING, ROOT` but only one `String pathPrefix`

File: s3/src/main/java/no/unit/nva/s3/S3Driver.java [hunk 2]
```diff
@@ -151,8 +160,18 @@ private static void verifyThatRequiredEnvVariablesAreInPlace() {
...
 
-    private Path filenameForZippedFile() {
-        return Path.of(UUID.randomUUID().toString() + GZIP_ENDING);
+    private Path filenameForZippedFile(Path s3Folder) {
+        String folderPath = processPath(s3Folder);
+        return Path.of(folderPath, UUID.randomUUID().toString() + GZIP_ENDING);
+    }
+
+    private String processPath(Path s3Folder) {
+        String unixPath = s3Folder.toString()
```
Comment (torarnet) [inline s3/src/main/java/no/unit/nva/s3/S3Driver.java:169]: Unix is AWS style, right? It should not be a problem when developing on Windows machines?

**Review by torarnet — APPROVED** [review]

File: s3/src/test/java/no/unit/nva/s3/S3DriverTest.java (hunk 1 above)
Comment (axthosarouris) [reply s3/src/test/java/no/unit/nva/s3/S3DriverTest.java:188]: These are the values the parameter will take successively

File: s3/src/main/java/no/unit/nva/s3/S3Driver.java (hunk 2 above)
Comment (axthosarouris) [reply s3/src/main/java/no/unit/nva/s3/S3Driver.java:169]: There is no local implementation of S3

**Pull request merged**